api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -436,4 +363,65 @@ private DbConceptSet getDbConceptSet(
     }
     return conceptSet;
   }
+
+  public static ConceptSet toClientConceptSet(DbConceptSet dbConceptSet) {
+    return new ConceptSet()
+        .etag(Etags.fromVersion(dbConceptSet.getVersion()))
+        .lastModifiedTime(dbConceptSet.getLastModifiedTime().getTime())
+        .creationTime(dbConceptSet.getCreationTime().getTime())
+        .description(dbConceptSet.getDescription())
+        .id(dbConceptSet.getConceptSetId())
+        .name(dbConceptSet.getName())
+        .domain(dbConceptSet.getDomainEnum())
+        .participantCount(dbConceptSet.getParticipantCount())
+        .survey(dbConceptSet.getSurveysEnum())
+        .creator(
+            dbConceptSet.getCreator() != null ? dbConceptSet.getCreator().getUsername() : null);
+  }
+
+  private ConceptSet toClientConceptSetWithConcepts(DbConceptSet dbConceptSet) {
+    ConceptSet conceptSet = toClientConceptSet(dbConceptSet);
+    if (!dbConceptSet.getConceptIds().isEmpty()) {
+      Iterable<DbConcept> concepts = conceptService.findAll(dbConceptSet.getConceptIds());
+      conceptSet.setConcepts(
+          Streams.stream(concepts)
+              .map(ConceptsController::toClientConcept)
+              .sorted(CONCEPT_NAME_ORDERING)
+              .collect(Collectors.toList()));
+    }
+    return conceptSet;
+  }
+
+  private DbConceptSet fromClientConceptSet(CreateConceptSetRequest request, long workspaceId) {
+    ConceptSet conceptSet = request.getConceptSet();
+    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
+    DbConceptSet dbConceptSet = new DbConceptSet();
+    dbConceptSet.setDomainEnum(conceptSet.getDomain());
+    if (conceptSet.getSurvey() != null) {
+      dbConceptSet.setSurveysEnum(conceptSet.getSurvey());
+    }
+    if (dbConceptSet.getDomainEnum() == null) {
+      throw new BadRequestException(
+          "Domain " + conceptSet.getDomain() + " is not allowed for concept sets");
+    }
+    if (conceptSet.getEtag() != null) {
```

Comment:
nit: `you can also do Optional.ofNullable().ifPresent(etag -> setEtagStuff(etag));`.